packages/shared/src/components/fields/DragDrop.tsx

```diff
@@ -0,0 +1,382 @@
+  /** Maximum file size in MB (convenience prop) */
+  maxSizeMB?: number;
+  /** Allowed file types (MIME types) */
+  acceptedTypes?: string[];
+  /** Allowed file extensions */
+  acceptedExtensions?: string[];
+  /** Maximum number of files allowed */
+  maxFiles?: number;
+  /** Whether to allow multiple files */
+  multiple?: boolean;
+}
+
+export interface DragDropError {
+  type: 'size' | 'format' | 'count' | 'unknown';
+  message: string;
+  file?: File;
+}
+
+export interface DragDropProps {
+  /** Callback when files are dropped/selected */
+  onFilesDrop: (files: File[], errors?: DragDropError[]) => void;
+  /** Validation rules */
+  validation?: DragDropValidation;
+  /** Custom error messages */
+  errorMessages?: {
+    size?: string;
+    format?: string;
+    count?: string;
+    unknown?: string;
+  };
+  /** Whether the component is disabled */
+  disabled?: boolean;
+  /** Custom className for the drop zone */
+  className?: string;
+  /** Content to show when no files are selected */
+  children?: ReactNode;
+  state?: MutationStatus;
+  inputRef?: MutableRefObject<HTMLInputElement>;
+  isCompactList?: boolean;
+  renameFileTo?: string;
+  hiddenInput?: boolean;
+}
+
+const BYTES_PER_MB = 1024 * 1024;
+
+const defaultErrorMessages = {
+  size: 'File size exceeds the maximum allowed size',
+  format: 'File type is not supported',
+  count: 'Too many files selected',
+  unknown: 'An error occurred while processing the file',
+};
+
+const getIcon = (state: MutationStatus) => {
+  if (state === 'pending') {
+    return <Loader />;
+  }
+
+  if (state === 'success') {
+    return <ChecklistAIcon className="text-accent-avocado-default" />;
+  }
+
+  return null;
+};
+
+interface ItemProps {
+  name: string;
+  state: MutationStatus;
+  uploadAt?: Date;
+}
+
+const LargeItem = ({ name, state, uploadAt }: ItemProps) => (
+  <div className="flex w-full items-center gap-1">
+    <DocsIcon secondary size={IconSize.Size48} />
+    <div className="flex min-w-0 flex-1 flex-col">
+      <Typography className="truncate" type={TypographyType.Subhead} bold>
+        {name}
+      </Typography>
+      {uploadAt && (
+        <Typography
+          className="truncate"
+          type={TypographyType.Caption2}
+          color={TypographyColor.Tertiary}
+          bold
+        >
+          {uploadAt.toDateString()}
+        </Typography>
+      )}
+    </div>
+    {getIcon(state)}
+  </div>
+);
+
+const CompactItem = ({ name, state }: ItemProps) => (
+  <div className="flex w-full items-center gap-1">
+    <DocsIcon secondary />
+    <div className="min-w-0 flex-1 text-left">
+      <Typography className="truncate" type={TypographyType.Footnote} bold>
+        {name}
+      </Typography>
+    </div>
+    {getIcon(state)}
+  </div>
+);
+
+export function DragDrop({
+  onFilesDrop,
+  validation = {},
+  errorMessages = {},
+  disabled = false,
+  className,
+  children,
+  state,
+  inputRef: inputRefProps,
+  isCompactList,
+  renameFileTo,
```

Comment:
for now I don't think we'll need to rename files. Wdyt?

Reply:
This was only for display purposes. Nothing functional to the BE.